integration_tests/testsuite/test_util.py

```diff
@@ -156,6 +160,60 @@ def retrieve_url_for_version(version):
         app_dict = json.loads(subprocess.check_output(url_command))
         return app_dict.get('versionUrl')
     except (subprocess.CalledProcessError, ValueError, KeyError) as e:
-        logging.warn('Error encountered when retrieving app URL! %s', e)
-        sys.exit(1)
+        logging.error('Error encountered when retrieving app URL! %s', e)
+        raise e
     raise Exception('Unable to contact deployed application!')
+
+
+def generate_gke_image_name():
+    return 'gcr.io/{project}/{image}'.format(
+        project=project_id(),
+        image=_generate_hex_token(8)
+    )
+
+
+def generate_gke_service_name():
+    return 'gcp-integration-test-{0}'.format(_generate_hex_token(8))
+
+
+def generate_namespace():
+    return 'int-test-{0}'.format(_generate_hex_token(8))
+
+
+@retry(wait_exponential_multiplier=1000, wait_exponential_max=32000,
+       stop_max_attempt_number=12)
+def get_external_ip_for_cluster(service_name, namespace):
```

Comment:
If this takes awhile, there are a few other options. You can hit the IP of any node in the cluster with a service type of NodePort.